File: problems/2264-largest-3samedigit-number-in-string/solution.cpp
class Solution {
public:
    string largestGoodInteger(string num) {
        string ans = "";
        for (int i = 0; i < num.length() - 2; ++i) {
            if (num[i] == num[i + 1] && num[i + 1] == num[i + 2]) {
                string current = num.substr(i, 3);
                if (ans == "" || current > ans) {
                    ans = current;
                }
            }
        }
        return ans;
    }
};

File: problems/2264-largest-3samedigit-number-in-string/solution.java
class Solution {
    public String largestGoodInteger(String num) {
        String maxGoodInteger = "";
        for (int i = 0; i <= num.length() - 3; i++) {
            String sub = num.substring(i, i + 3);
            if (sub.charAt(0) == sub.charAt(1) && sub.charAt(1) == sub.charAt(2)) {
                if (maxGoodInteger.compareTo(sub) < 0) {
                    maxGoodInteger = sub;
                }
            }
        }
        return maxGoodInteger;
    }
}